.github/workflows/test.yml

```diff
@@ -0,0 +1,186 @@
+name: 'Run tests'
+
+on:
+  pull_request:
+    branches-ignore:
+      - 'docker'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      ################################################################
+      #################   INSTALL DEPENDENCIES  ######################
+      ################################################################
+
+      - name: Install dependencies
+        run: |
+          sudo apt update
+          sudo apt install graphviz bc
+          pip install requests numpy matplotlib pycodestyle
+
+      - name: Fetch flexfringe binary
+        env:
+          GH_TOKEN: ${{ github.token }}
+          FF_REPO: "https://github.com/tudelft-cda-lab/FlexFringe"
+        shell: bash
+        run: |
+          gh release download latest -R $FF_REPO -p "flexfringe-x64-linux"  # Might have to be updated if FlexFringe decides to change the binary name
```

Comment:
I have added comments like this to indicate the dependency on FlexFringe (also present in other parts of this file)